pac4j-core/src/main/java/org/pac4j/core/credentials/authenticator/LocalCachingAuthenticator.java

```diff
@@ -56,10 +59,34 @@ public void validate(final T credentials, final WebContext context) throws HttpA
             });
             credentials.setUserProfile(profile);
             logger.debug("Found cached credential. Using cached profile {}...", profile);
-        } catch (Exception e) {
-            throw new CredentialsException(e);
+        } catch (ExecutionException e) {
+            // this covers all the checked exception of validate
+            if (e.getCause() instanceof HttpAction) {
+                throw (HttpAction) e.getCause();
+            } else {
+                throw rethrowWrapper(e);
+            }
+        } catch (UncheckedExecutionException e) {
+            if (e.getCause() instanceof RuntimeException) {
+                throw (RuntimeException) e.getCause();
```

Comment:
(Note that normally you can simply add commits to my branch, if you want to, according to the way github works)